### PR TITLE
Update/home fragment

### PR DESCRIPTION
**Pull request by ghost**
fragments w/o query code

### CODE REVIEW

File: app/src/main/java/com/fbu/fbuteam/activities/HomeActivity.java [hunk 1]
```diff
@@ -1,18 +1,102 @@
...
 import com.fbu.fbuteam.R;
+import com.fbu.fbuteam.fragments.NewsFragment;
+import com.fbu.fbuteam.fragments.ProfileFragment;
+import com.google.android.material.bottomnavigation.BottomNavigationView;
 
 public class HomeActivity extends AppCompatActivity {
+    private Toolbar toolbar;
+    private BottomNavigationView bnvHome;
+    static MenuItem miActionProgressItem;
+
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/activities/HomeActivity.java]: two empty lines here

File: app/src/main/java/com/fbu/fbuteam/activities/HomeActivity.java [hunk 2]
```diff
@@ -1,18 +1,102 @@
...
 
 import com.fbu.fbuteam.R;
+import com.fbu.fbuteam.fragments.NewsFragment;
+import com.fbu.fbuteam.fragments.ProfileFragment;
+import com.google.android.material.bottomnavigation.BottomNavigationView;
 
 public class HomeActivity extends AppCompatActivity {
+    private Toolbar toolbar;
+    private BottomNavigationView bnvHome;
+    static MenuItem miActionProgressItem;
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/activities/HomeActivity.java]: change to `private static` and change variable name to `menuActionProgressItem` would be better

File: app/src/main/java/com/fbu/fbuteam/activities/NewsAdapter.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        private void bind(final Node node) {
+            //TODO: bind the view elements to the news article node
+        }
+    }
+
+    public void clear () {
+        newsNodes.clear();
+        notifyDataSetChanged();
+    }
+
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/activities/NewsAdapter.java]: you can remove this space here

File: app/src/main/java/com/fbu/fbuteam/activities/NewsAdapter.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        private TextView tvTitle;
+        private TextView tvCreatedAt;
+        private TextView tvBodySnippet;
+
+        private ViewHolder(View itemView) {
+            super(itemView);
+            tvTitle = itemView.findViewById(R.id.tvTitle);
+            tvCreatedAt = itemView.findViewById(R.id.tvCreatedAt);
+            tvBodySnippet = itemView.findViewById(R.id.tvBodySnippet);
+
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/activities/NewsAdapter.java]: you can remove this space here

File: app/src/main/java/com/fbu/fbuteam/fragments/NewsFragment.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+import com.fbu.fbuteam.utils.EndlessRecyclerViewScrollListener;
+import com.fbu.fbuteam.activities.HomeActivity;
+import com.fbu.fbuteam.activities.NewsAdapter;
+
+import org.w3c.dom.Node;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/NewsFragment.java]: three empty lines...

File: app/src/main/java/com/fbu/fbuteam/fragments/NewsFragment.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;
+
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/NewsFragment.java:14]: two empty lines...

File: app/src/main/java/com/fbu/fbuteam/fragments/NewsFragment.java [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    private RecyclerView rvNews;
+    private NewsAdapter adapter;
+    private List<Node> myNewsNodes;
+    private LinearLayoutManager linearLayoutManager;
+    public static final int newsNodesPerQuery = 20;
+    private SwipeRefreshLayout swipeContainer;
+    private EndlessRecyclerViewScrollListener scrollListener;
+
+
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/NewsFragment.java]: two empty lines...

File: app/src/main/java/com/fbu/fbuteam/fragments/NewsFragment.java [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+
+
+public class NewsFragment extends Fragment {
+
+    private RecyclerView rvNews;
+    private NewsAdapter adapter;
+    private List<Node> myNewsNodes;
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/NewsFragment.java]: only `news` is better, I think

File: app/src/main/java/com/fbu/fbuteam/fragments/NewsFragment.java [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+    private LinearLayoutManager linearLayoutManager;
+    public static final int newsNodesPerQuery = 20;
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/NewsFragment.java]: put the static variable on the top of the Class

File: app/src/main/java/com/fbu/fbuteam/fragments/NewsFragment.java [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_news, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/NewsFragment.java]: empty lines here

File: app/src/main/java/com/fbu/fbuteam/fragments/NewsFragment.java [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+        swipeContainer = view.findViewById(R.id.swipeContainer);
+    }
+
+    private void endlessScrollListener() {
+        scrollListener = new EndlessRecyclerViewScrollListener(linearLayoutManager) {
+            @Override
+            public void onLoadMore(int page, int totalItemsCount, RecyclerView view) {
+                loadNextData();
+            }
+
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/NewsFragment.java]: empty lines here

File: app/src/main/java/com/fbu/fbuteam/fragments/NewsFragment.java [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    private void loadNextData() {
+        //TODO query for next posts for endless scroll
+    }
+
+    private void queryForNodes() {
+        //TODO query for top news articles
+    }
+
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/NewsFragment.java]: you can remove this empty space here

File: app/src/main/res/layout/item_news_node.xml [hunk 13]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout
+    xmlns:android="http://schemas.android.com/apk/res/android" android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools">
+
+
```
Comment (LeoNero) [inline app/src/main/res/layout/item_news_node.xml]: two empty lines

File: app/src/main/res/layout/item_news_node.xml [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+        android:layout_marginEnd="8dp"
+        android:layout_marginRight="8dp"
+        android:text="TextView"
+        app:layout_constraintBottom_toTopOf="@+id/imageView"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.497"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/tvTitle" />
+
+
```
Comment (LeoNero) [inline app/src/main/res/layout/item_news_node.xml]: two empty lines

**Review by LeoNero — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/fbu/fbuteam/utils/EndlessRecyclerViewScrollListener.java [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+
+    public void resetState() {
+        this.currentPage = this.startingPageIndex;
+        this.previousTotalItemCount = 0;
+        this.loading = true;
+    }
+
+    public abstract void onLoadMore(int page, int totalItemsCount, RecyclerView view);
+
+
```
Comment (haleyharelson) [inline app/src/main/java/com/fbu/fbuteam/utils/EndlessRecyclerViewScrollListener.java:92]: just fix some spacing

File: app/src/main/res/values/strings.xml [hunk 16]
```diff
@@ -12,4 +12,7 @@
     <string name="password">Password</string>
 
     <string name="create_account_message">Don\'t have an account? Sign up.</string>
+
```
Comment (haleyharelson) [inline app/src/main/res/values/strings.xml:15]: spacing

**Pull request merged**